app/src/main/java/com/duckduckgo/app/onboarding/OnboardingFlowCheckerImpl.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.onboarding
+
+import com.duckduckgo.app.cta.db.DismissedCtaDao
+import com.duckduckgo.app.cta.model.CtaId
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.ExtendedOnboardingFeatureToggles
+import com.duckduckgo.app.settings.db.SettingsDataStore
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class OnboardingFlowCheckerImpl @Inject constructor(
+    private val dismissedCtaDao: DismissedCtaDao,
+    private val extendedOnboardingFeatureToggles: ExtendedOnboardingFeatureToggles,
+    private val settingsDataStore: SettingsDataStore,
+) : OnboardingFlowChecker {
+
+    override fun isOnboardingComplete(): Boolean {
```

Comment:
As commented, this is not representing the onboarding is completed. It's only checks the Home CTAs onboarding related.

We could keep the method name but then we should include more checks, including one using `UserStage` to confirm is onboarding done.